src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```diff
@@ -0,0 +1,155 @@
+package org.jenkinsci.plugins.edithelp;
+
+import java.io.*;
+
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.RootAction;
+import hudson.model.View;
+import jenkins.model.Jenkins;
+import org.apache.tools.ant.taskdefs.Recorder;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.lang.Klass;
+import sun.misc.URLClassPath;
+
+import javax.swing.*;
+import java.net.Socket;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+@Extension
+public class EditHelpRootAction implements RootAction {
+
+    //reading the catalog of helpmanager
+    private static ArrayList<File> listWithFileNames = new ArrayList<>();
+    public static void getListFiles(String str) {
+        File f = new File(str);
+        File[] mydr = f.listFiles();
+        if (mydr == null)
+            return;
+        for (File s : mydr) {
+            if (s.isFile()) {
+                listWithFileNames.add(s);
+            } else if (s.isDirectory()) {
+                getListFiles(s.getAbsolutePath());
+            }
+        }
+    }
+
+    //array for cache storage
+    Map<String, String> map = new TreeMap<String, String>();
+
+    //reading files into cache
+    public EditHelpRootAction() throws IOException, FileNotFoundException {
+        Jenkins jn = Jenkins.getInstance();
+        if (jn != null) {
```

Comment:
This is a root action and it can be assumed that `getInstance()` returns an instance rather than `null`.